core/src/test/java/org/apache/calcite/test/RelBuilderTest.java

```diff
@@ -3440,10 +3440,14 @@ private void checkExpandTable(RelBuilder builder, Matcher<RelNode> matcher) {
                 builder.field("DNAME"),
                 builder.field("DEPTNO"))
             .build();
+    final String expectedTree = ""
+        + "LogicalSort(sort0=[$1], sort1=[$0], dir0=[ASC], dir1=[ASC])\n"
+        + "  LogicalValues(tuples=[[]])\n";
+    assertThat(root, hasTree(expectedTree));
     try (PreparedStatement preparedStatement = RelRunners.run(root)) {
-      final String s = CalciteAssert.toString(preparedStatement.executeQuery());
-      final String result = "";
-      assertThat(s, is(result));
+      final String result = CalciteAssert.toString(preparedStatement.executeQuery());
+      final String expectedResult = "";
+      assertThat(result, is(expectedResult));
```

Comment:
This is not expected. That means the rule doesn't work.

Reply:
I mean the plan.

Reply:
This is the expected plan right built by the RelBuilder, before any PruneEmptyRules is applied.

Reply:
I guess we only care about plan generated by planner, not the builder.

Reply:
OK, I found that it is not able to compare the final plan here.

Reply:
I will add another test in RelOptRulesTest to check the plan before and after the rule is applied